lambda/lambda.ml

```diff
@@ -673,79 +673,101 @@ let rec make_sequence fn = function
    Assumes that the image of the substitution is out of reach
    of the bound variables of the lambda-term (no capture). *)
 
-let subst update_env s lam =
-  let rec subst s lam =
-    let remove_list l s =
-      List.fold_left (fun s (id, _kind) -> Ident.Map.remove id s) s l
-    in
+let subst update_env ?(on_bound_variable = fun id -> id) s lam =
```

Comment:
`on_bound_variable` should probably be a `freshen_bound_variables` (a boolean). I can't see a good use of a generic function here that is not `Ident.rename`

Reply:
Done.